rexfw/statistics/writers/graphite.py

```diff
@@ -0,0 +1,76 @@
+from pickle import dumps
+from struct import pack
+from socket import socket
+
+# timestamp is the number of seconds since unix epoch time.
+# Carbon-cache will use the time of arrival if the timestamp is set to -1.
+# https://graphite.readthedocs.io/en/stable/feeding-carbon.html
+
+
+class GraphiteStatisticsWriter:
+    """
+    Writes statistics to Graphite server.
+    :param str job_name
+    :param str graphite_url
+    :param int graphite_port
+    """
+
+    def __init__(self, job_name, graphite_url="127.0.0.1", graphite_port=2004):
+
+        self.job_name = job_name
+        self.sock = socket()
+        self.sock.connect((graphite_url, graphite_port))
+
+    def make_tuples_list(self, step, elements):
+        # export list of (path, (timestamp, value)). for more info look at:
+        # https://graphite.readthedocs.io/en/stable/feeding-carbon.html
+        return []
+
+    def write(self, step, elements):
+        listOfStatsTuples = self.make_tuples_list(step, elements)
+        payload = dumps(listOfStatsTuples, protocol=2)
+        size = pack("!L", len(payload))
+        self.sock.sendall(size)
+        self.sock.sendall(payload)
+        # TODO: for large size of data, they should be pushed once a certain number of samples
+        # (METRICS_BUFFER_SIZE) are buffered or upon a defined period of time (MAX_BUFFER _TIME)
+        # For more info look at
+        # https://www.metricfire.com/blog/monitoring-with-graphite-installation-and-setup/
+
+
+class GraphiteREStatisticsWriter(GraphiteStatisticsWriter):
+    def make_tuples_list(self, step, elements):
+        timestamp = step
+        listOfStatsTuples = []
+        elements = [e for e in elements if e.name == "acceptance rate"]
+        for e in elements:
+            if len(e.origins) == 1:
```

Comment:
These elements would need to go in the MCMC statistics writer below.
Sorry if I haven't been clear about that; it's quite confusing to have two "acceptance rate" quantities :disappointed: but the one with `len(e.origins) == 1` is the acceptance rate of a _local_ MCMC sampler and thus needs to be in the other class, while the acceptance rates with `len(e.origins) == 2` are the acceptance rates for exchanges between replicas and stay in this class.